doc/devel/documenting_mpl.rst

```diff
@@ -29,349 +29,170 @@ requirements that are needed to build the documentation.  They are listed in
   * `Graphviz <http://www.graphviz.org/Download.php>`_ is not a Python package,
     and needs to be installed separately.
 
-General structure
------------------
+General file structure
+----------------------
 
 All documentation is built from the :file:`doc/` directory.  This directory
-contains both ``.rst`` files that contain pages in the documentation,
-directories that contain more ``.rst`` files, and configuration files for
-Sphinx_.
+contains both ``.rst`` files that contain pages in the documentation and
+configuration files for Sphinx_.
+
+The ``.rst`` files are kept in :file:`doc/users`,
+:file:`doc/devel`, :file:`doc/api` and :file:`doc/faq`. The main entry point is
+:file:`doc/index.rst`, which pulls in the :file:`index.rst` file for the users
+guide, developers guide, api reference, and FAQs. The documentation suite is
+built as a single document in order to make the most effective use of cross
+referencing.
 
 .. note::
 
-   An exception to this are the directories :file:`gallery` and
+   An exception to this are the directories :file:`examples` and
    :file:`tutorials`, which exist in the root directory.  These contain Python
    files that are built by `Sphinx Gallery`_.  When the docs are built,
-   directories of the same name will be generated inside of :file:`docs/`.  The
-   generated directories :file:`docs/gallery` and :file:`docs/tutorials` can be
-   safely deleted.
+   the directories :file:`docs/gallery` and :file:`docs/tutorials`
+   are generated.
```

Comment:
I'd say something like:  "are automatically generated.  Do not edit the rst files in :file:`docs/gallery` and :file:`docs/tutorials`, as they are rebuilt from the sources in the root directory"

I was mystified for a while when my changes to files in these directories were being overwritten ;-)